kcqrs-appengine/src/main/kotlin/com/clouway/kcqrs/adapter/appengine/AppEngineEventStore.kt

```diff
@@ -84,11 +110,23 @@ class AppEngineEventStore(private val kind: String = "Event", private val messag
 
             aggregateEntity.setUnindexedProperty(eventsProperty, aggregateEvents)
             aggregateEntity.setUnindexedProperty(versionProperty, currentVersion + events.size)
-            dataStore.put(transaction, listOf(snapshot, aggregateEntity))
-
-            transaction.commit()
+            return try {
+                dataStore.put(transaction, listOf(snapshotEntity, aggregateEntity))
+                transaction.commit()
+                SaveEventsResponse.Success(aggregateId, currentVersion)
+            } catch (ex: ApiProxy.RequestTooLargeException) {
```

Comment:
You can check the size of the entity by converting it to Protocol Buffer. This is going to skip the round trip to the datastore. 
```kotlin
val ep = EntityTranslator.convertToPb(entity)
ep.propertySize()
```

Currently entity size is limited to: 1,048,572 bytes

Reply:
Agreed will be implemented.

Reply:
Sadly i don't think this is a proper way to check size as the propertySize returns int and it is 1. This is actually some list size in the context of 
```Kotlin
OnestoreEntity.Property
```
which is a list and apparently this returns the size of that list.

Reply:
The following snippet returns the proper size of the encoded protocol buffer:

```kotlin
  entity = Entity("test")
    entity.setProperty("test1", "test")
    entity.setProperty("test2", "teaskdjakdjajsdst2")
    entity.setProperty("test3", "test3")
val pb = EntityTranslator.convertToPb(entity)
    println(pb.serializedSize)
```

The size is changed as I change the content or the number of properties.

Reply:
I was trying with .propertySize(). Will test with serializedSize.